Common/Data/Custom/SmartInsider/SmartInsiderTransaction.cs

```diff
@@ -0,0 +1,374 @@
+                NextCloseBegin = NextCloseBegin,
+                LastCloseEnded = LastCloseEnded,
+                SecurityDescription = SecurityDescription,
+                TickerCountry = TickerCountry,
+                TickerSymbol = TickerSymbol,
+                AnnouncementDate = AnnouncementDate,
+                TimeReleased = TimeReleased,
+                TimeProcessed = TimeProcessed,
+                TimeReleasedUtc = TimeReleasedUtc,
+                TimeProcessedUtc = TimeProcessedUtc,
+                AnnouncedIn = AnnouncedIn,
+
+                BuybackDate = BuybackDate,
+                BuybackVia = BuybackVia,
+                BuybackBy = BuybackBy,
+                HoldingType = HoldingType,
+                Currency = Currency,
+                Price = Price,
+                TransactionAmount = TransactionAmount,
+                GBPValue = GBPValue,
+                EURValue = EURValue,
+                USDValue = USDValue,
+                NoteText = NoteText,
+                BuybackPercentage = BuybackPercentage,
+                VolumePercentage = VolumePercentage,
+                ConversionRate = ConversionRate,
+                AmountAdjustedFactor = AmountAdjustedFactor,
+                PriceAdjustedFactor = PriceAdjustedFactor,
+                TreasuryHolding = TreasuryHolding,
+
+                Symbol = Symbol,
+                Value = Value,
+                Time = Time
+            };
+        }
+
+        /// <summary>
+        /// Converts the data to CSV
+        /// </summary>
+        /// <returns>String of CSV</returns>
+        /// <remarks>Parsable by the constructor should you need to recreate the object from CSV</remarks>
+        public override string ToCsv()
+        {
+            return $"{TransactionID}\t{BuybackType}\t{LastUpdate:yyyyMMdd}\t{LastIDsUpdate:yyyyMMdd}\t{ISIN}\t{USDMarketCap}\t{CompanyID}\t{ICBIndustry}\t{ICBSuperSector}\t{ICBSector}\t{ICBSubSector}\t{ICBCode}\t{CompanyName}\t{PreviousResultsAnnouncementDate:yyyyMMdd}\t{NextResultsAnnouncementsDate:yyyyMMdd}\t{NextCloseBegin:yyyyMMdd}\t{LastCloseEnded:yyyyMMdd}\t{SecurityDescription}\t{TickerCountry}\t{TickerSymbol}\t{AnnouncementDate:yyyyMMdd}\t{TimeReleased:yyyyMMdd HH:mm:ss}\t{TimeProcessed:yyyyMMdd HH:mm:ss}\t{TimeReleasedUtc:yyyyMMdd HH:mm:ss}\t{TimeProcessedUtc:yyyyMMdd HH:mm:ss}\t{AnnouncedIn}\t{BuybackDate:yyyyMMdd}\t{BuybackVia}\t{BuybackBy}\t{HoldingType}\t{Currency}\t{Price}\t{TransactionAmount}\t{GBPValue}\t{EURValue}\t{USDValue}\t{NoteText}\t{BuybackPercentage}\t{VolumePercentage}\t{ConversionRate}\t{AmountAdjustedFactor}\t{PriceAdjustedFactor}\t{TreasuryHolding}";
+        }
+
+        /// <summary>
+        /// Determines equality to another SmartInsiderTransaction instance
+        /// </summary>
+        /// <param name="other">Another SmartInsiderTransaction instance</param>
+        /// <returns>Boolean value indicating equality</returns>
+        public override bool Equals(SmartInsiderEvent other)
```

Comment:
Another thought -- most `BaseData` implementations _do not_ override the `Equals` method. The general thinking is that `BaseData` instances have _reference identity_ and therefore do _not_ use value-type equality semantics but rather reference type equality semantics. A notable exception is the `IndicatorDataPoint`.

What was the motivation for overriding `Equals` in these types?

Reply:
It's legacy code that I intended on using to get rid of duplicates, but decided that comparing the strings was a better way of making sure no duplicates existed in new additions to the file